vector/src/main/java/im/vector/app/features/onboarding/OnboardingViewModel.kt

```diff
@@ -581,42 +591,77 @@ class OnboardingViewModel @AssistedInject constructor(
-        val homeServerConnectionConfig = homeServerConnectionConfigFactory.create(homeserverUrl)
+    private fun handleHomeserverChange(action: OnboardingAction.HomeServerChange) {
+        val homeServerConnectionConfig = homeServerConnectionConfigFactory.create(action.homeServerUrl)
         if (homeServerConnectionConfig == null) {
             // This is invalid
             _viewEvents.post(OnboardingViewEvents.Failure(Throwable("Unable to create a HomeServerConnectionConfig")))
         } else {
-            startAuthenticationFlow(homeServerConnectionConfig)
+            startAuthenticationFlow(action, homeServerConnectionConfig)
         }
     }
 
-    private fun startAuthenticationFlow(homeServerConnectionConfig: HomeServerConnectionConfig, serverTypeOverride: ServerType? = null) {
+    private fun startAuthenticationFlow(trigger: OnboardingAction?, homeServerConnectionConfig: HomeServerConnectionConfig, serverTypeOverride: ServerType? = null) {
         currentHomeServerConnectionConfig = homeServerConnectionConfig
 
         currentJob = viewModelScope.launch {
             setState { copy(isLoading = true) }
-
             runCatching { startAuthenticationFlowUseCase.execute(homeServerConnectionConfig) }.fold(
-                    onSuccess = {
-                        rememberHomeServer(homeServerConnectionConfig.homeServerUri.toString())
-                        if (it.isHomeserverOutdated) {
-                            _viewEvents.post(OnboardingViewEvents.OutdatedHomeserver)
-                        }
+                    onSuccess = { onAuthenticationStartedSuccess(trigger, homeServerConnectionConfig, it, serverTypeOverride) },
+                    onFailure = { _viewEvents.post(OnboardingViewEvents.Failure(it)) }
+            )
+            setState { copy(isLoading = false) }
+        }
+    }
 
-                        setState {
-                            copy(
-                                    serverType = alignServerTypeAfterSubmission(homeServerConnectionConfig, serverTypeOverride),
-                                    selectedHomeserver = it.selectedHomeserver,
-                                    isLoading = false,
-                            )
+    private suspend fun onAuthenticationStartedSuccess(
+            trigger: OnboardingAction?,
+            config: HomeServerConnectionConfig,
+            authResult: StartAuthenticationFlowUseCase.StartAuthenticationResult,
+            serverTypeOverride: ServerType?
+    ) {
+        rememberHomeServer(config.homeServerUri.toString())
+        if (authResult.isHomeserverOutdated) {
+            _viewEvents.post(OnboardingViewEvents.OutdatedHomeserver)
+        }
+
+        when (trigger) {
+            is OnboardingAction.HomeServerChange.EditHomeServer   -> {
+                when (awaitState().onboardingFlow) {
+                    OnboardingFlow.SignUp -> internalRegisterAction(RegisterAction.StartRegistration) { _ ->
```

Comment:
adding the `internalRegisterAction` to the edit server flow is the fix for the missing errors, updating the server selection after success also ensures when closing the screen the previous homeserver selection in not overwritten